app/Http/Controllers/Api/RequestApiController.php

```diff
@@ -77,34 +73,57 @@ public function storeRequest(Request $request): Response
     }
 
     /**
-     * @param  mixed  $resource
-     * @param  null|string  $format
-     * @return \App\Http\Resources\Html\Request|\App\Http\Resources\Json\Request|\App\Http\Resources\Xml\Request
+     * Validate storeRequest.
+     *
+     * @param Request $request
      */
-    protected function singleResourceResponse($resource, ?string $format)
+    private function validateStoreRequest(Request $request)
     {
-        if ($format == static::FORMAT_HTML) {
-            return new \App\Http\Resources\Html\Request($resource);
-        } elseif ($format == static::FORMAT_XML) {
-            return new \App\Http\Resources\Xml\Request($resource);
-        } else {
-            return new \App\Http\Resources\Json\Request($resource);
-        }
+        $this->validate($request, [
+            'url' => [
+                'required',
+                'url',
+            ],
+            'method' => [
+                'required',
+                'in:'.implode(',', \App\Request::getAllowedMethods()),
+            ],
+            'profile' => [
+                'required',
+                'exists:profiles,identifier',
+            ],
+            'request_headers' => 'array',
+            'request_headers.*.name' => [
+                'required',
+                'string',
+                'max:255',
+            ],
+            'request_headers.*.value' => [
+                'required',
+                'string',
+                'max:16777215',
+            ],
+        ]);
     }
 
     /**
-     * @param  mixed  $resources
-     * @param  null|string  $format
-     * @return \App\Http\Resources\Html\RequestCollection|\App\Http\Resources\Json\RequestCollection|\App\Http\Resources\Xml\RequestCollection
+     * Create EndpointRequest for Request and Endpoint.
+     *
+     * @param Request $request
+     * @param Endpoint $endpoint
+     * @return EndpointRequest
      */
-    protected function resourceCollectionResponse($resources, ?string $format)
+    private function createEndpointRequest(Request $request, Endpoint $endpoint): EndpointRequest
     {
-        if ($format == static::FORMAT_HTML) {
-            return new \App\Http\Resources\Html\RequestCollection($resources);
-        } elseif ($format == static::FORMAT_XML) {
-            return new \App\Http\Resources\Xml\RequestCollection($resources);
-        } else {
-            return new \App\Http\Resources\Json\RequestCollection($resources);
-        }
+        $profile = Profile::whereIdentifier($request->get('profile'))
```

Comment:
What happens if no profile is found? Does the following code break? 😬

Reply:
The profile is "found" in the validation before creating (and whering (lol) the profile).